### ISSUE TITLE
Option to hide the badge count of unread messages in muted chats

### ISSUE THREAD

**Issue by tonysung** [peter-iakovlev/Telegram#31]
This is already done on desktop per https://github.com/overtake/telegram/issues/141 .

Should do the same on iOS.

**Comment by 4tuneTeller** [comment]
+1 I'd sugest and option to completely disable notifications from muted chats (badge numbers are also set by push notifications, if I'm not mistaken), so they won't drain my battery.

**Comment by etonline** [comment]
Is it the default for muted supergroup? on my device, muted supergroup doesn't count into badge numbers, and I actually want it to be counted... :(

**Comment by dotKirk** [comment]
+1 Dear god please. I've had to leave so many groups because I can't handle the ever-increasing badge count.

**Comment by stepan001** [comment]
+1

**Comment by audiocommander** [comment]
+1

Doesn't look really complicated: https://github.com/peter-iakovlev/Telegram/blob/58ce610e95aec809a847e928e06c01cffbd317a1/Telegraph/TGAppDelegate.mm#L797

It's already been implemented in the OS X Version:
https://github.com/overtake/telegram/issues/141#

**Comment by SomniusTiger** [comment]
+1 from me as well. I like having the notification bubble on iOS for things that still need my attention. If I've muted a chat, I've decided that unless I'm notified directly (via reply or mention) that I don't need to pay attention to every unread message for that chat. Without this feature, the notification bubble no longer has any importance, and I can't tell if I need to reply to someone or not.

**Comment by Fenrikur** [comment]
+1
Being able to mute chats without also having them excluded from the unread badge counter makes the badge rather useless indeed, especially since the functionality itself is already available in clients on for other platforms as mentioned above.

**Comment by last-chance** [comment]
+1 

Telegram has quickly replaced text messaging as my way of communicating with friends and family, but now one of the volunteer organizations I help run has decided to use Telegram for communication. 

Now my badge counter is permanently in the high hundreds, and I can't really tell when someone is actually trying to get ahold of me. This is getting to a point where I may have to stop using Telegram for anything important.

**Comment by ha33an** [comment]
i have same problem
correct that

**Comment by r3nya** [comment]
+1

**Comment by Gryzor** [comment]
+1000 :)

**Comment by pinver** [comment]
+1

**Comment by gustavoavena** [comment]
+1

**Comment by barsukov** [comment]
It is still the issue here :( I mean I still has a problem with it and to be honest I have no idea why it is so complicated to implement so useful bug.

**Comment by mrsyasi** [comment]
+1

**Comment by mlabarca** [comment]
Definitely +1

**Comment by adabu** [comment]
+1

**Comment by mdevaev** [comment]
+1

**Comment by maxdrift** [comment]
+1

**Comment by alexinside** [comment]
+1 
_It is very annoying._ Already thinking to either uninstall the app or disable Badge Icon (iOS). Both solutions are bad.

**Comment by barsukov** [comment]
I am simply disabled the badge for app. We will see hopefully they will fix it for ios10 update.

**Comment by iamDecode** [comment]
+1, please, this is currently making Telegram unusable for me..

**Comment by iamDecode** [comment]
+1, This is currently making Telegram unusable for me...

**Comment by r3nya** [comment]
@iamDecode and all … Now everything is ok for me. Please check. :)

**Comment by iamDecode** [comment]
@r3nya What do you mean? Version 3.13 (current App Store version) nor this repo have solved this issue.

**Comment by audiocommander** [comment]
@r3nya nice that everything is ok for you, I can't see any update or fix or new setting.
I basically stopped using telegram for other communications than group chats and disabled the badge setting. So, currently Telegram is one of the most unreliable ways to contact me and the reason is this  annoying issue. Telegram would be the best messenger ever if it just got the notifications right. I mean, notifications are one of the most essential aspects of a messenger...

**Comment by r3nya** [comment]
Ah, damn … Sorry guys. I mean version for iOS.
@iamDecode @audiocommander

**Comment by audiocommander** [comment]
yes, we're talking about the iOS version? as iamDecode said, there's no fix - or do you have different informations?

**Comment by r3nya** [comment]
@audiocommander yes. I has a lot of muted chats … And my telegram icon without badge. Strange. :)

**Comment by pochemuto** [comment]
@r3nya so have I. I had received a lot of messages today and badges and notifications appeared only from unmated chats. 
iOS version 3.13

**Comment by r3nya** [comment]
Ha! Gotcha! I can't see the badge for muted chats, but I have badge for bots. But I disabled notifications for this bots. 😞 

// cc @audiocommander

**Comment by audiocommander** [comment]
I would rather think that this is another bug. Just made a test. My badge showed 15 unread messages, but I am actually having only 2 unmuted messages and 335 muted messages from various groups. After launching Telegram and not reading any new message, the badge dropped to 14 unread.
That **definitely** doesn't work.
(latest iOS, latest Telegram, iPhone 6)

**Comment by sdugoten** [comment]
Would that be possible that making a switch on badge count for each group? So, one mute switch and one badge switch for each group.  So, I could still mute the channel but somehow know someone is talking in a group when badge count shows. BTW, this is IOS version

**Comment by pdostal** [comment]
It'd be very nice and it's even solved for [OSX client](https://github.com/overtake/telegram/issues/141).

**Comment by CareF** [comment]
Also expect this feature

**Comment by alexdiment** [comment]
I'm surprised this takes forever to fix.

**Comment by mikexine** [comment]
Looking forward to this :)

**Comment by ralphbert** [comment]
+1

**Comment by nezaidu** [comment]
+111111111

**Comment by emieljanson** [comment]
+1

**Comment by rivf** [comment]
+1

**Comment by z3t0** [comment]
This needs to be implemented on the server side AFAIK which won't happen unless either telegram decides to work on it or releases the source code. Correct me if I am wrong but the source code for the server is not open source?

**Comment by iamDecode** [comment]
Why? This was already once implemented on the desktop app (before the introduction of the web app) and worked fine. The client just has to determine what unread counts to sum into the final app badge, I don't see how the server is involved in this at all.

**Comment by Gryzor** [comment]
I agree with @iamDecode 

App Badges are locally implemented by Cocoa/UIKit, I don't see how it has anything to do with the server. The app receives messages (among a gazillion other things) via the API, the status of a Chat is locally known by the App, and so the decision to create the Badge *should* rely on client code given the current API.

I know this is possible because the Mac app was doing it and it was using the Public API.

**Comment by z3t0** [comment]
@Gryzor @iamDecode Sorry I was referring to iOS where the badge count is directly connected to the APN push servers. Mac Desktop already has this feature so not sure why that's even a concern.

![](https://i.imgur.com/FKC4n9r.png)

> This is already done on desktop per overtake/telegram#141 .
Should do the same on iOS.


As far as I can tell this issue is for iOS

**Comment by RoOb17** [comment]
+1
I have a muted supergroup and it's not showing in the badge while the muted group is showing

**Comment by z3t0** [comment]
Just FYI.

For more of an explanation as to why this needs to be done on the server side.

The way notifications work on mobile devices when the app is not on the foreground is that push notifications are sent to the servers connected to the phone, in this case Apple's push servers. One of the pieces of information sent is the number of unread notifications.

Currently, on iOS the server sends all the unread count, and does not remove the muted unread count. This is not an issue on Desktop because the server ends it directly to the app which can easily hide any unread messages for the muted converations.

AFAIK telegrams server code is not accessible nor is it open source so unless they fix this I dont really know what there is to be done.

**Comment by JulyIghor** [comment]
+100

**Comment by sergioamr** [comment]
+1

**Comment by z3t0** [comment]
This issue should be locked as I have mentioned above that the issue needs to be resolved by telegram and cannot be fixed here.

**Comment by leo150** [comment]
Please fix this issue, it's very annoying

**Comment by Gryzor** [comment]
@leo150 Did you actually read the comments above you before posting?
@z3t0 So are you receiving a plain notification with a number and nothing else? Wouldn’t It be possible (albeit lame) to locally track, which ones are muted and subtract from that number? don’t you have to then show the number of unread messages in each group/chat?

**Comment by z3t0** [comment]
@Gryzor The notification isn't shown in the notifications bar if muted, but it is shown in the badge count.

For example if I mute chat A but not chat B then I only see notifications for chat B in the notification bar as expected. BUT notifications from chatA still add to the badge count on the homescreen, which is unexpected.

The problem is that mobile operating systems (especially iOS) are very aggressive about allowing background processes to run which is why it would not be possible to implement a local solution. The only way to "solve" it locally is to adjust the badge count once Telegram is opened, which is not ideal as the whole point of a badge count is to get and idea of notifications without opening the app.

This isn't a problem on a desktop as it can just run its own local process instead of relying on the badge count number being sent from the server.

**Comment by Gryzor** [comment]
@z3t0 As an iOS/Android dev myself, I totally get what you’re saying. Until the server allows the clients to send the notification preferences per-chat, there’s nothing to do as it is.

**Comment by z3t0** [comment]
This again brings up the issue of having a partially open source project...

**Comment by Gryzor** [comment]
@z3t0 Don’t open (no pun intended?) that _can of worms_ ;)

**Comment by bun4uk** [comment]
+1

**Comment by hashier** [comment]
@z3t0 @Gryzor Well that is actually not 100 % true what you are saying. The iOS app in fact has a chance to intercept notifications send to the device and modify them (since iOS 10).
You are even allowed to make network requests during updating the notifications.

Prominent apps that use this are encrypted messenger apps, that send the encrypted message as a payload to the phone and the phone decrypts the message on the device, updates the notification and then displays it to the user.

Obviously it still isn't easy to go through all the muted channels, update them and then subtract that number from the badge, but it is possible.

I quickly wrote a test app, [gist here](https://gist.github.com/hashier/cc09d1c5bcfc99d4e283b34b2aa8e826)

 https://www.xkcd.com/386/

**Comment by z3t0** [comment]
Thanks for the information! This looks quite promising.

**Comment by kathampy** [comment]
+1

**Comment by themotu** [comment]
Why can't we just have a "disable badge" per channel?

**Comment by dmitriyabr** [comment]
Any news, guys?

**Comment by z3t0** [comment]
@dmitriyabr I think it all depends on whether a dev is willing to spend the time to implement this feature since @hashier has pointed out that it should be technically possible.

**Comment by ghost** [comment]
So strange - i've got 5 chats, all of them were muted. But i see badge count for one of them. Hm. I tried to mute this chat from phone and from os x client - nothing changes - badge still appears.

**Comment by dmitriyabr** [comment]
@sacred0x01 
four chats is a supergroups
one is not

**Comment by ghost** [comment]
@dmitriyabr оh, thx a lot. I have some ideal like your, but with proof it looks better)

**Comment by the-eugen** [comment]
+1

**Comment by DimICE** [comment]
+1

**Comment by SleepingWhale** [comment]
+1

**Comment by elreymon** [comment]
+1

**Comment by monomadic** [comment]
How is this not a thing? Why is telegram implementing crazy video-stickers, themes, and all kinds of advanced features without a simple customisation that would actually improve our lives?

Like, I'm a telegram zealot, but honestly this one small thing is ruining my life. Dramatic, no? haha... what I mean is... some things in life should be 'pull' instead of 'push', some groups that while I love to be in, don't warrant a beep every 10 seconds interrupting my life... Having a big red badge on certain groups when I'm checking actual important messages is SOUL DRAINING... to the point that I just uninstalled telegram from everything but my backup phone so I could ignore that badge... (hopefully nothing happens to that phone because I'll be locked out of telegram thanks to its poorly-thought-out SMS root-level-verification, and I don't have SMS.. but that's a separate issue).

PLEASE implement this, it's driving me crazy.

ps. it's two months later since I did this, still only have telegram on a backup phone. I HIGHLY recommend it until this 'life experience bug' is fixed :P I have peace again

**Comment by pdostal** [comment]
Well, true. There're groups I want to be part of but I need to constantly scroll them to get rid of 'unread messages'.

**Comment by ehsemsar** [comment]
+1

**Comment by Grinv** [comment]
+1

**Comment by nicolinux** [comment]
I wonder how many +1 it takes for this to become a thing without some snarky dev suggesting "If you want it, do it yourself..."...

**Comment by pandrez** [comment]
+1

**Comment by evasyuk** [comment]
+1

**Comment by YNedderhoff** [comment]
+1

**Comment by f1ght4fun** [comment]
is there anyone really looking at this request - it is obvious thing present on desktop app and not implemented for over a year for mobile platforms! please address it if possible!

**Comment by Leukodi** [comment]
+1

**Comment by monomadic** [comment]
Coming soon in 2024 - introducing telegram 20.5!!! - with new snapchat filters for your dog and cat, and the ability to teleport food to your coworkers!!! But sorry we still didn't fix that major obvious notifications issue from way back in 2017. Maaaaaybe next time.

**Comment by f1ght4fun** [comment]
New update with photos grouping etc and still no option migrated from desktop to mobile settings which is the most commented and requested for such a long time .... unbelievable ... is there anyone alive there from a support team to reply to all these people ?

**Comment by xzaleksey** [comment]
+1

**Comment by timurztd** [comment]
+1

**Comment by maxkolenko** [comment]
+1
Please

**Comment by undr26** [comment]
+1

**Comment by ottodevs** [comment]
Is the worst anti usability feature in this software, for the sake of god merge this PR.
Has no sense to have 3000 unread notifications when I never asked nor wanted to have these notifications or even when I have those groups muted in purpose.

**Comment by squaredcircle** [comment]
+1

I receive a red unread counter badge on both the Telegram iOS icon and on the “Chats” tab in the client. This number currently includes a count from muted supergroups, which it shouldn’t. Muted regular groups do not appear in the tally of the red unread counter badge. The supergroups themselves appear in client with a grey unread counter badge.

Please fix, this issue has been annoying me for well over a year.

**Comment by zeonardo** [comment]
+1

**Comment by DeJordev** [comment]
+1

**Comment by SleepingWhale** [comment]
Telegram X doesn't have this problem.

**Comment by f1ght4fun** [comment]
@SleepingWhale indeed !!! great, tnx a lot for the suggestion!

**Comment by zolotokrylin** [comment]
+1
Need to move to another app, as this really makes my experience bad

**Comment by realdavidvega** [comment]
Totally agree.

**Comment by slavikme** [comment]
+1
Pretty annoying bug. As a workaround, I need constantly convince group admins to upgrade the group to supergroup for this to work.

**Comment by MasoudRahmani** [comment]
+1
Please hide it in ios

**Comment by sergei-iurchenko** [comment]
+1

**Comment by gfelipe** [comment]
+1

**Comment by SanzharK** [comment]
+1, please give us the option to turn the count off for muted groups

**Comment by Schtolc** [comment]
+1

**Comment by multeg** [comment]
+1

**Comment by anisimovdk** [comment]
+1

**Comment by shoce** [comment]
+1

**Comment by Oneiros90** [comment]
+1

**Comment by turnmusic** [comment]
+1

**Comment by tru2dagame** [comment]
+1

**Comment by oh6fub** [comment]
Just switched to iOS. This is really awkward not having the the option to leave out notifications from muted chats! And since this issue was opened two years ago already, seems there's lack of interest to implement this feature.

+1 to suggestion anyway

**Comment by kastal** [comment]
+1
The numbers get so high, that it can't even display the whole number, tens of thousands. What's the point of that?

**Comment by joolsmtw** [comment]
+1 Please do it, it's driving me crazy. And it's not hard to do.

**Comment by AntMartDev** [comment]
+1
Please implement this, it's so annoying to not be able to tell if I need to read a message or not...

**Comment by scarlehoff** [comment]
If Telegram wants to be an usable app in ios, it should fix this. It’s stupid.

**Comment by ce1** [comment]
Not sure if it's just me but there is an option in settings to "Include muted
chats in unread count" which is enabled by default. If you unselect this, seems
to be working fine.  





On Thu, May 3, 2018 5:26 PM, Antonio Martinelli notifications@github.com  wrote:
+1
Please implement this, it's so annoying to not be able to tell if I need to read
a message or not...

**Comment by scarlehoff** [comment]
@ce1 where exactly do you find this option? 
I’m looking in settings > notificationd and sounds but cannot see it?

**Comment by joolsmtw** [comment]
Deleted my original +1, because it seemed to be fixed.
The setting should be able to be found under Settings > Notifications as indicated above.

https://imgur.com/a/mjv3Grm - here is a screenshot of what I see.

**Comment by Gryzor** [comment]
@joolsmtw Do you realize this is about Telegram for **iOS** and not **macOS**, right? The shot you posted is from the Desktop Client.

**Comment by joolsmtw** [comment]
@Gryzor tru. My bad.

**Comment by jpottaway** [comment]
Thought I was going mad and missing a feature! This thread is a relief and a massive frustration. 2 years!! Def only effects some muted groups, as previously mentioned groups/supergroups. Makes the badge useless once a critical mass of muted groups is reached. 

Hate to say it, but Hangouts does it better. (Shudder)

+1

**Comment by mogunov** [comment]
+1

**Comment by alexshpilkin** [comment]
Please refrain from leaving ±1 comments with no substantial argumentation; use the GitHub [emoji reactions][1] instead. Otherwise, bugmail becomes very spammy. (I know that this comment will contribute to the spamminess as well. Sorry.)

[1]: https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/

**Comment by jerrygreen** [comment]
Is this a problem for 2 years? Lol. I cant use these chats because of this

Unread quantity digit is too annoying, should be able to be disabled

**Comment by oh6fub** [comment]
I switched to Telegram X too. Not sure what the downsides will be, but at least I got rid of this annoyance.

**Comment by LanKing** [comment]
+1

**Comment by sidarali** [comment]
+1

**Comment by vitfury** [comment]
Unusable application. After seeing this two year old issue with so many pluses, i’ve decided to quit telegram.

**Comment by diegoadrada** [comment]
Still nothing, this is really really annoying.

**Comment by pedrotambo** [comment]
> Still nothing, this is really really annoying.

As ce1 already said, this problem has been solved. Settings > Include muted chats in unread count -> Uncheck

**Comment by gfelipe** [comment]
> > Still nothing, this is really really annoying.
> 
> As ce1 already said, this problem has been solved. Settings > Include muted chats in unread count -> Uncheck

Only thing that is the option on the desktop client, this thread is about the iPhone (iOS) client. Still no way to do that

**Comment by YuppY** [comment]
It was fixed with latest update 5.0.8:

![Telegram screenshot](https://i.imgur.com/u6CghyG.png)

Telegram X has this option too:

![Telegram X screenshot](https://i.imgur.com/RkPKPyd.png)

**Comment by jerrygreen** [comment]
Hooray, they eventually did this!

P.S. Waited a year for this ;)
